tests/runtime/engine/runner.py

```diff
@@ -320,11 +327,20 @@ def run_test(test):
                 print('\tCLEANUP error: %s' % e.stderr)
                 return Runner.FAIL
 
+        def print_befores_output():
+            if len(befores_output) > 0:
+                print("\tBefore cmd output:")
+                for (out, err) in befores_output:
+                    out = out.decode("utf-8")
+                    err = err.decode("utf-8")
+                    print(f"stdout: {out}\nstderr stderr: {err}")
+
         if p and p.returncode != 0 and not test.will_fail and not timeout:
             print(fail("[  FAILED  ] ") + "%s.%s" % (test.suite, test.name))
-            print('\tCommand: ' + bpf_call)
+            print('\tCommand: ' + (bpf_call or "unset"))
             print('\tUnclean exit code: ' + str(p.returncode))
             print('\tOutput: ' + output.encode("unicode_escape").decode("utf-8"))
+            print_befores_output()
```

Comment:
This looks like a useful feature which is not exactly specific to this PR. Would you mind moving it into a separate PR, preferably with something similar for the `AFTER` directive?